lesion_analysis/time_point_lesion_evolution.py

```diff
@@ -0,0 +1,417 @@
+"""
+This python file is used to analyse the evolution of the lesions over time.
+It first performs a registration of the images and their segmentation on the two time points.
+Then, it performs clustering of the lesions on each image and computes the volume and the center of each lesion. 
+Finally, it compares the lesions between the two time points and computes the evolution of the lesions.
+
+Args:
+    -i-1, --input-first_image: path to the first image 
+    -i-2, --input-second_image: path to the second image
+    -seg-1, --segmentation-first_image: path to the lesion segmentation of the first image
+    -seg-2, --segmentation-second_image: path to the lesion segmentation of the second image
+    -o, --output_folder: path to the output folder
+    --plot: whether to plot the results or not
+
+Returns:
+    None
+
+Example:
+    python time_point_lesion_evolution.py -i-1 path/to/first/image -i-2 path/to/second/image -seg-1 path/to/first/segmentation -seg-2 path/to/second/segmentation -o path/to/output/folder
+    python3 lesion_analysis/time_point_lesion_evolution.py -i-1 /Users/plbenveniste/Desktop/lesion_comparison_copy/sub-cal072_ses-M0_STIR.nii.gz -i-2 /Users/plbenveniste/Desktop/lesion_comparison_copy/sub-cal072_ses-M12_STIR.nii.gz -seg-1 /Users/plbenveniste/Desktop/lesion_comparison_copy/sub-cal072_ses-M0_STIR_lesion-manual.nii.gz -seg-2 /Users/plbenveniste/Desktop/lesion_comparison_copy/M12_inference_results.nii.gz -o /Users/plbenveniste/Desktop/lesion_comparison_copy/output
+
+To do:
+    * Remove some of the outputed files
+
+Pierre-Louis Benveniste
+"""
+
+import os
+import argparse
+from pathlib import Path
+import nibabel as nib
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.cluster import DBSCAN
+import networkx as nx
+from scipy.optimize import linear_sum_assignment
+
+def get_parser():
+    """
+    This function parses the arguments given to the script.
+
+    Args:
+        None
+
+    Returns:
+        parser: parser containing the arguments
+    """
+
+    parser = argparse.ArgumentParser(description='Analyse the results of the segmentation of the MS lesion on the spinal cord.')
+    parser.add_argument('-i-1', '--input-first_image', required=True,
+                        help='Path to the first image')
+    parser.add_argument('-i-2', '--input-second_image', required=True,
+                        help='Path to the second image')
+    parser.add_argument('-seg-1', '--segmentation-first_image', required=True,
+                        help='Path to the lesion segmentation of the first image')
+    parser.add_argument('-seg-2', '--segmentation-second_image', required=True,
+                        help='Path to the lesion segmentation of the second image')
+    parser.add_argument('-o', '--output_folder', required=True,
```

Comment:
suggestion: input the SC segmentation instead of computing it here-- why? because users can QC the segs and make sure they are fine (otherwise, if seg is failing, the registration will fail, blablabla)